Lib/test/test_importlib/test_util.py

```diff
@@ -757,5 +757,49 @@ def test_source_from_cache_path_like_arg(self):
  ) = util.test_both(PEP3147Tests, util=importlib_util)
 
 
+class MagicNumberTests:
+    """
+    Test release compatibility issues relating to importlib
+    """
+    def test_magic_number(self):
+        """
+        Each python minor release should generally have a MAGIC_NUMBER
+        that does not change once the release reaches candidate status.
+
+        Once a release reaches candidate status, the value of the constant
+        EXPECTED_MAGIC_NUMBER in this test should be changed.
+        This test will then check that the actual MAGIC_NUMBER matches
+        the expected value for the release.
+
+        In exceptional cases, it may be required to change the MAGIC_NUMBER
+        for a maintenance release. In this case the change should be
+        discussed in python-dev. If a change is required, community
+        stakeholders such as OS package maintainers must be notified
+        in advance. Such exceptional releases will then require an
+        adjustment to this test case.
+        """
+        if sys.version_info.releaselevel not in ('final', 'candidate'):
+            return
+        EXPECTED_MAGIC_NUMBER = 3379
+        actual = int.from_bytes(self.util.MAGIC_NUMBER[:2], 'little')
+
+        msg = (
+            "Candidate and final releases require the current  "
```

Comment:
I'd start this message with "To avoid breaking backwards compatibility with cached bytecode files that can't be automatically regenerated by the current user, ..."